src/sdk/modules/validators/smartSessionValidator/tosmartSessionValidatorModule.ts

```diff
@@ -0,0 +1,118 @@
+ * @param client The client instance.
+ * @param initData Initialization data for the module.
+ * @param deInitData De-initialization data for the module.
+ * @returns A promise that resolves to a Smart Session Validator Module instance.
+ *
+ * @example
+ * const module = await toSmartSessionValidatorModule({
+ *   nexusAccountAddress: '0x1234...',
+ *   client: nexusClient,
+ *   initData: '0x...',
+ *   deInitData: '0x...'
+ * });
+ *
+ * // Use the module
+ * const dummySignature = await module.getStubSignature();
+ * const userOpSignature = await module.signUserOpHash('0x...');
+ * const messageSignature = await module.signMessage('Hello, world!');
+ */
+export const toSmartSessionValidatorModule = async ({
+  nexusAccountAddress,
+  initData,
+  deInitData,
+  client,
+  activePermissionId = "0x"
+}: {
+  nexusAccountAddress: Hex
+  initData: Hex
+  deInitData: Hex
+  client: Client
+  activePermissionId: Hex
+}): Promise<ToSmartSessionValidatorModuleReturnType> => {
+  // Note: session key signer (applies in case of K1 based simple session validator algorithm)
+  const signer = await toSigner({ signer: client.account as Account })
+
+  return toValidationModule({
+    address: addresses.SmartSession,
+    nexusAccountAddress,
+    initData,
+    deInitData,
+    activePermissionId,
+    getStubSignature: async (
+      moduleSignatureMetadata?: ModuleSignatureMetadata
+    ) => {
+      // console.log("activePermissionId", activePermissionId)
+      const signature = encodeSmartSessionSignature({
+        mode: moduleSignatureMetadata?.mode
+          ? moduleSignatureMetadata.mode
+          : SmartSessionMode.USE,
+        permissionId: moduleSignatureMetadata?.permissionId
+          ? moduleSignatureMetadata.permissionId
+          : activePermissionId,
+        signature: DUMMY_ECDSA_SIG,
+        enableSessionData: moduleSignatureMetadata?.enableSessionData
+      }) as Hex
+      return signature
+    },
+    signUserOpHash: async (
+      userOpHash: Hex,
+      moduleSignatureMetadata?: ModuleSignatureMetadata
+    ) => {
+      const signature = await signer.signMessage({
+        message: { raw: userOpHash as Hex }
+      })
+      const moduleSignature = encodeSmartSessionSignature({
+        mode: moduleSignatureMetadata?.mode
+          ? moduleSignatureMetadata.mode
+          : SmartSessionMode.USE,
+        permissionId: moduleSignatureMetadata?.permissionId
+          ? moduleSignatureMetadata.permissionId
+          : activePermissionId,
+        signature: signature,
+        enableSessionData: moduleSignatureMetadata?.enableSessionData
+      }) as Hex
+      return moduleSignature
+    },
+    signMessage: async (_message: Uint8Array | string) => {
+      const message =
+        typeof _message === "string" ? _message : { raw: _message }
+      let signature = await signer.signMessage({ message })
+      const potentiallyIncorrectV = Number.parseInt(signature.slice(-2), 16)
```

Comment:
Let's move this chunk of code to a helper. It's currently in about 6 different files

Reply:
I see it at 3 places

Reply:
made helper sanitizeSignature